src/MICore/UnixUtilities.cs

```diff
@@ -42,6 +42,12 @@ public static class UnixUtilities
             string pidFifo,
             string debuggerCmd)
         {
+            // On OSX, 'wait' will return once there is a status change from the launched process rather than for it to exit, so
+            // we need to use 'fg' there. 
+            // On Linux, fg will fail with 'no job control' because our bash shell is not interactive, so we need 
+            // to use 'wait' there.
+            string waitForCompletionCommand = PlatformUtilities.IsOSX() ? "fg > /dev/null; " : "wait $pid; ";
```

Comment:
nit: You might want to move this to a `static readonly` field, but I don't really have a strong opinion on it.